File: src/site/wenlong/dimens/core/DimensConverter.java
package site.wenlong.dimens.core;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.PlatformDataKeys;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.Messages;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;
import site.wenlong.dimens.form.ConverterDialog;
import site.wenlong.dimens.languages.LanguagesFactory;
import site.wenlong.dimens.utils.LegalUtils;

import static site.wenlong.dimens.core.Configuration.PLUGINS_NAME;

/**
 * Dimens转换插件的Action
 *
 * @author : 郭文龙
 * @date : 2020/6/22  17:30
 */
public class DimensConverter extends AnAction {
    private final Configuration mConfiguration = Configuration.getInstance();

    @Override
    public void actionPerformed(@NotNull AnActionEvent event) {
        VirtualFile currentFile = event.getData(PlatformDataKeys.VIRTUAL_FILE);
        Project project = event.getData(PlatformDataKeys.PROJECT);
        ConverterDialog converterDialog = null;
        if (LegalUtils.isLayoutFolder(currentFile)) {
            converterDialog = new ConverterDialog(currentFile, project, true);
        } else if (LegalUtils.isLayoutXml(currentFile)) {
            converterDialog = new ConverterDialog(currentFile, project, false);
        } else {
            Messages.showMessageDialog(LanguagesFactory.createText(mConfiguration.languageIndex).getLayoutTipsErrorFile(), PLUGINS_NAME, Messages.getInformationIcon());
        }
        if (converterDialog != null) {
            converterDialog.setSize(640, 480);
            converterDialog.setLocationRelativeTo(null);
            converterDialog.setVisible(true);
        }
    }
}

File: src/main/kotlin/site/wenlong/dimens/action/converter/DimensConverter.kt
package site.wenlong.dimens.action.converter

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.PlatformDataKeys
import com.intellij.openapi.ui.Messages
import com.intellij.openapi.vfs.VirtualFile
import site.wenlong.dimens.base.BaseAnAction
import site.wenlong.dimens.constant.Constant
import site.wenlong.dimens.ext.showErrorMessage
import site.wenlong.dimens.languages.LanguagesFactory

/**
 * 转换layout.xml或layout文件夹下所有layout的action
 *
 * @author : 郭文龙
 * @Email  : guowenlong20000@sina.com
 * @date   : 2023/5/6  20:52
 */
class DimensConverter : BaseAnAction() {
    override fun actionPerformed(event: AnActionEvent) {
        val currentFile = event.getData(PlatformDataKeys.VIRTUAL_FILE) ?: throw Exception("plugins compatibility error")
        val project = event.getData(PlatformDataKeys.PROJECT) ?: throw Exception("plugins compatibility error")
        when {
            isLayoutFolder(currentFile) -> {
                DimensConverterDialog(currentFile, project, true).showFrame()
            }

            isLayoutXml(currentFile) -> {
                DimensConverterDialog(currentFile, project, false).showFrame()
            }

            else -> {
                showErrorMessage(LanguagesFactory.createText(configuration.languageIndex).layoutTipsErrorFile)
            }
        }
    }

    /**
     * 是否是layout文件夹
     */
    private fun isLayoutFolder(currentFile: VirtualFile?): Boolean {
        return currentFile != null && currentFile.name == Constant.LAYOUT_FOLDER_NAME
    }

    /**
     * 是否是layout.xml文件
     */
    private fun isLayoutXml(currentFile: VirtualFile?): Boolean {
        return currentFile != null && currentFile.name.endsWith(Constant.LAYOUT_XML_NAME)
    }
}